src/OrchardCore.Modules/OrchardCore.Queries/Sql/SqlParser.cs

```diff
@@ -246,20 +246,36 @@ private void EvaluateOrderClause(ParseTreeNode parseTreeNode)
         var idList = parseTreeNode.ChildNodes[2];
 
         _modes.Push(FormattingModes.SelectClause);
+
         for (var i = 0; i < idList.ChildNodes.Count; i++)
         {
-            var id = idList.ChildNodes[i].ChildNodes[0];
-
             if (i > 0)
             {
                 _builder.Append(", ");
             }
 
+            var id = idList.ChildNodes[i].ChildNodes[0];
+
+            // RANDOM() is a special case where we need to use the dialect's random function.
+            if (id.ChildNodes[0].Token?.ValueString == "RANDOM")
```

Comment:
This could work in SQLite & Postgres only, `RAND` is the one that works in SQL Server and MySQL

Reply:
`_builder.Append(_dialect.RandomOrderByClause);`

But maybe you forgot how sql queries are managed in this feature. Call me if you want a quick refresher.